circuit_maintenance_parser/parser.py

```diff
@@ -37,10 +37,25 @@ def get_data_types(cls) -> List[str]:
         """Return the expected data type."""
         return cls._data_types
 
-    def parse(self, raw: bytes) -> List[Dict]:
-        """Extract a list of data that partially or completely describes a series of Maintenance objects."""
+    def parser_hook(self, raw: bytes):
+        """Custom parser logic."""
         raise NotImplementedError
 
+    def parse(self, raw: bytes) -> List[Dict]:
+        """Execute parsing."""
+        try:
+            result = self.parser_hook(raw)
+            if result == [{}]:
```

Comment:
Kind of a weird special case, at a glance. Should we instead check for `result == []` or `(not item for item in result)` to check for the case where no data at all is returned and/or the case where multiple datas are returned but they're all empty?